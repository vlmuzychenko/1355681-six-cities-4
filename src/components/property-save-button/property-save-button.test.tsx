import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import NameSpace from "../../reducer/name-space.js";
import PropertySaveButton from "./property-save-button";

const mockStore = configureStore([]);

const AuthorizationStatusMock = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const cardClassNameMock = `property__bookmark-button`;
const iconClassNameMock = `property__bookmark-icon`;
const widthMock = `31`;
const heightMock = `33`;

it(`Should PropertySaveButton render correctly`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatusMock.NO_AUTH,
      authorizationData: {},
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <PropertySaveButton
            className={cardClassNameMock}
            iconClassName={iconClassNameMock}
            width={widthMock}
            height={heightMock}
            isFavorite={true}
            authStatus={AuthorizationStatusMock.AUTH}
            offerId={1}
          />
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

