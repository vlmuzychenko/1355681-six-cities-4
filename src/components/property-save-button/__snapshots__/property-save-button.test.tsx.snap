// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PropertySaveButton render correctly 1`] = `
<button
  className="property__bookmark-buttonproperty__bookmark-button property__bookmark-buttonproperty__bookmark-button--active button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="property__bookmark-icon"
    height="33"
    width="31"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In
     bookmarks
  </span>
</button>
`;
